File: Framework/tolua/typeof.lua
local var_0_0 = type
local var_0_1 = {}
local var_0_2 = tolua.typeof
local var_0_3 = tolua.findtype

function typeof(arg_1_0)
	local var_1_0 = var_0_0(arg_1_0)
	local var_1_1

	if var_1_0 == "table" then
		var_1_1 = var_0_1[arg_1_0]

		if var_1_1 == nil then
			var_1_1 = var_0_2(arg_1_0)
			var_0_1[arg_1_0] = var_1_1
		end
	elseif var_1_0 == "string" then
		var_1_1 = var_0_1[arg_1_0]

		if var_1_1 == nil then
			var_1_1 = var_0_3(arg_1_0)
			var_0_1[arg_1_0] = var_1_1
		end
	else
		error(debug.traceback("attemp to call typeof on type " .. var_1_0))
	end

	return var_1_1
end

File: Framework/tolua/typeof.py
from luatable import table

var_0_1 = table()
import findtype #????

def typeof(arg_1_0):
	var_1_0 = type(arg_1_0)

	if var_1_0 == table:
		var_1_1 = var_0_1[arg_1_0]

		if var_1_1 == None:
			var_1_1 = typeof(arg_1_0)
			var_0_1[arg_1_0] = var_1_1
	elif var_1_0 == str:
		var_1_1 = var_0_1[arg_1_0]

		if var_1_1 == None:
			var_1_1 = findtype(arg_1_0)
			var_0_1[arg_1_0] = var_1_1
	else:
		raise ValueError(f"attempt to call typeof on type {var_1_0})")

	return var_1_1
